### ISSUE TITLE
Natural Cure revealed upon a status switch out

### ISSUE THREAD

**Issue by Layell** [Zarel/Pokemon-Showdown#1452]
%Marty: this is kind of related: Natural Cure actually reveals itself in-game when something statused with Natural Cure switches out
%Marty: the Poke Ball goes from yellowish to normal

**Comment by Zarel** [comment]
Which gens?

**Comment by Marty-D** [comment]
All gens.

Gen 3 has the opposing party lineup appear for a few seconds just before a Pokemon is sent out for you to keep track of: http://i.imgur.com/lECVyeS.png
Gen 4 has the same thing, but it's also on the bottom screen while you select an action between turns: http://i.imgur.com/mA6qNIO.png
Gen 5 only has it on the bottom screen between turns.
Gen 6 only has it on the top screen between turns.

Basically, if you're paying attention it's easy to notice when something that's statused and switches out has Natural Cure because as soon as you see the lineup again there will be one less darkened or yellow ball. This can also be useful for Soundproof information after a Heal Bell in Gen 3 and 4.

**Comment by Layell** [comment]
Doubles/Triples would also make this a bit more complicated if two Pokemon could possibly have Natural Cure and a double switch is performed. In that case Natural Cure is revealed on the switch back in instead. But in singles is 100% is confirmed on switch out.

**Comment by ascriptmaster** [comment]
Server: `|-activate|ability: Natural Cure`

Client: `(A Pokemon was cured of its status with Natural Cure.)`

**Comment by Marty-D** [comment]
Even in Doubles for Gen 3 and 4 Natural Cure would be obvious during a double switch since the ball lineup shows up moments before each switch-in happens. But yeah, for Gen 5 and 6 it's more complicated since you have to wait until you can choose actions to see it.

**Comment by Zarel** [comment]
>Client: `(A Pokemon was cured of its status with Natural Cure.)`

Why not name the Pokémon?

**Comment by Slayer95** [comment]
Sometimes you can't know which Pokémon it is.
e.g: In doubles, if both active foes are statused, they both switch out, but only one of the replacement Pokémon is healthy.

**Comment by ascriptmaster** [comment]
For gen 3-4, we can have the Pokemon name since the ball lineup is shown briefly before a Pokemon switches in.

For gen 5+, we have to actually make it a residual effect because you won't know until the turn has ended! Quite annoying

**Comment by Slayer95** [comment]
Use case for Soundproof:
Suppose that I get to know that my opponent has two Electrodes with different movesets. One of them happens to have Soundproof as ability while the other one has Static, but I don't know that.

At some point of the battle, one of them switches in, gets statused and switches out after showing its unique moveset. On switch-out, the third Pokéball changes to "statused" mode. Later on, the healthy Electrode comes in, and also switches out after getting statused. On switch-out, the fourth Pokéball changes to "statused" mode.

Celebi switches in and uses Heal Bell. Only the fourth Pokéball gets cleared as a result. Under such situation, I would immediately gain knowledge of which Electrode is statused and which one isn't.

(@Marty-D, please correct me if I missed something in the scenario I described?)

Players in PS should be able to use this battle information too. It occurs to me that the only way to do this is by revealing which slots are healthy/statused/fainted as in-game. This conveys a big challenge for the battle UI, as it's not totally possible to associate the status ailments with foe Pokémon identities (e.g. when more than one Pokémon is healthy and, simultaneously, more than one Pokémon is statused).

**Comment by Zarel** [comment]
You know, order doesn't actually matter, as long as all the information is sent before the next decision.

**Comment by Marty-D** [comment]
Correct, @Slayer95. Here's a scenario taking your Illusion example a step further: http://i.imgur.com/CoetcRf.png

Your opponent leads with Hitmonlee and it uses a non-Zoroark move as you paralyze it.

Your opponent switches to Arbok and some stuff happens but eventually you throw out another paralysis-inducing move while for whatever reason they switch directly to Zoroark (which looks like Hitmonlee but you know it's Zoroark because you already paralyzed Hitmonlee).

Your opponent then switches to Druddigon. Now you're thinking, man Hitmonlee and Zoroark have the same status and neither has taken damage, how will I know which is which???

At this point if you aren't paying attention and you see a paralyzed Hitmonlee switch in safely later, you would actually have no way of knowing if it's Zoroark or the real Hitmonlee.

If you check out the ball positions though, you would know exactly where they are. Either (1) happens and it's clearly Zoroark or (2) happens and it's clearly Hitmonlee.

And yeah, like Slayer95 mentioned, at any point after turn 2, as long as Hitmonlee stays in the last slot, you know that nothing that switches in can be Zoroark unless it looks like Hitmonlee and isn't paralyzed.

**Comment by Slayer95** [comment]
@Zarel, what do you think about making the client side bars alternate between  a "Team Preview" display mode, with the Pokémon icons, and a "Side Status" display mode, with the Poké Ball icons?

**Comment by Zarel** [comment]
Why alternate? You can just add an icon for statuses. You can actually add the color for the exact status, like we do on the switch menu.

**Comment by Zarel** [comment]
Or at least that I was planning to do on the switch menu. idr if we ended up doing it.

**Comment by Slayer95** [comment]
The thing is that we can't unify the information in Doubles since BW.
If two statused Pokémon, among which exactly one has Natural Cure, switch-out in the same turn, and their replacements are healthy, the opponent can't know for sure which one is healthy and which one isn't.

**Comment by Zarel** [comment]
I guess handle that case separately.

**Comment by ascriptmaster** [comment]
Bump. What if we just had 6 mini-Pokeball icons in the trainer bars (or less if they have fewer Pokemon, which fixes another issue we have with formats with no Team Preview)? It's not perfect but at least diligent players will be able to notice the info.

- Gen 4 and below: Update on every switch-in
- Gen 5 and up: Update at end of every turn
